homeassistant/components/dyson/fan.py

```diff
@@ -70,40 +68,30 @@
 }
 
 
-SPEED_LIST_HA = [SPEED_LOW, SPEED_MEDIUM, SPEED_HIGH]
-
-SPEED_LIST_DYSON = [
-    int(FanSpeed.FAN_SPEED_1.value),
-    int(FanSpeed.FAN_SPEED_2.value),
-    int(FanSpeed.FAN_SPEED_3.value),
-    int(FanSpeed.FAN_SPEED_4.value),
-    int(FanSpeed.FAN_SPEED_5.value),
-    int(FanSpeed.FAN_SPEED_6.value),
-    int(FanSpeed.FAN_SPEED_7.value),
-    int(FanSpeed.FAN_SPEED_8.value),
-    int(FanSpeed.FAN_SPEED_9.value),
-    int(FanSpeed.FAN_SPEED_10.value),
+PRESET_MODE_AUTO = "auto"
+PRESET_MODES = [PRESET_MODE_AUTO]
+
+ORDERED_DYSON_SPEEDS = [
+    FanSpeed.FAN_SPEED_1,
+    FanSpeed.FAN_SPEED_2,
+    FanSpeed.FAN_SPEED_3,
+    FanSpeed.FAN_SPEED_4,
+    FanSpeed.FAN_SPEED_5,
+    FanSpeed.FAN_SPEED_6,
+    FanSpeed.FAN_SPEED_7,
+    FanSpeed.FAN_SPEED_8,
+    FanSpeed.FAN_SPEED_9,
+    FanSpeed.FAN_SPEED_10,
 ]
+DYSON_SPEED_TO_INT_VALUE = {k: int(k.value) for k in ORDERED_DYSON_SPEEDS}
+INT_VALUE_TO_DYSON_SPEED = {v: k for k, v in DYSON_SPEED_TO_INT_VALUE.items()}
 
-SPEED_DYSON_TO_HA = {
-    FanSpeed.FAN_SPEED_1.value: SPEED_LOW,
-    FanSpeed.FAN_SPEED_2.value: SPEED_LOW,
-    FanSpeed.FAN_SPEED_3.value: SPEED_LOW,
-    FanSpeed.FAN_SPEED_4.value: SPEED_LOW,
-    FanSpeed.FAN_SPEED_AUTO.value: SPEED_MEDIUM,
-    FanSpeed.FAN_SPEED_5.value: SPEED_MEDIUM,
-    FanSpeed.FAN_SPEED_6.value: SPEED_MEDIUM,
-    FanSpeed.FAN_SPEED_7.value: SPEED_MEDIUM,
-    FanSpeed.FAN_SPEED_8.value: SPEED_HIGH,
-    FanSpeed.FAN_SPEED_9.value: SPEED_HIGH,
-    FanSpeed.FAN_SPEED_10.value: SPEED_HIGH,
-}
+SPEED_LIST_DYSON = [int(speed) for speed in DYSON_SPEED_TO_INT_VALUE.values()]
```

Comment:
The items have already been copied to integer in the dict comprehension above.
```suggestion
SPEED_LIST_DYSON = [speed for speed in DYSON_SPEED_TO_INT_VALUE.values()]
```